### ISSUE TITLE
bad Boolean args not rejected

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#10337]
Booleans aren't detecting bad args well enough:

This is ok:

```
>>> I<2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\core\expr.py", line 299, in __lt__
    raise TypeError("Invalid comparison of complex %s" % me)
TypeError: Invalid comparison of complex I
```

This isn't:

```
>>> Interval(1,2)<3
False
>>> FiniteSet(2)<3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\sets\sets.py", line 1996, in __lt__
    return self.is_proper_subset(other)
  File "sympy\sets\sets.py", line 361, in is_proper_subset
    raise ValueError("Unknown argument '%s'" % other)
ValueError: Unknown argument '3'
```

**Comment by meghana1995** [comment]
I have added checks in the _lt and _gt functions in the sets module for bad args.

**Comment by oscarbenjamin** [comment]
Looks like this was supposed to be closed when #10342 got merged.
